app/main/views/g_cloud.py

```diff
@@ -116,13 +117,17 @@ def get_service_by_id(service_id):
             framework_slug = override_framework_slug
 
         framework = data_api_client.get_framework(framework_slug)['frameworks']
-        service_view_data = Service(
-            service_data,
-            content_loader.get_builder(framework_slug, 'display_service').filter(
-                service_data
-            ),
-            framework_helpers.get_lots_by_slug(framework)
-        )
+        try:
+            service_view_data = Service(
+                service_data,
+                content_loader.get_builder(framework_slug, 'display_service').filter(
+                    service_data
+                ),
+                framework_helpers.get_lots_by_slug(framework)
+            )
+        except ContentNotFoundError as e:
```

Comment:
Yep that feels like a better check doesn't it, Rob said a similar thing yesterday too.  If there's no manifest for a g-cloud framework then something is really wrong.